libs/blocks/modal/decorators/langnav.js

```diff
@@ -0,0 +1,37 @@
+import { getConfig } from '../../../utils/utils.js';
+
+export async function handleEvent(prefix, link, config) {
+  document.cookie = `international=${prefix};path=/`;
+  sessionStorage.setItem('international', prefix);
+  return fetch(link.href, { method: 'HEAD' }).then((resp) => {
+    if (!resp.ok) throw new Error('request failed');
+    window.location.assign(link.href);
+  }).catch(() => {
+    const prefixUrl = prefix ? `/${prefix}` : '';
+    window.location.assign(`${prefixUrl}${config.contentRoot || ''}/`);
+  });
+}
+
+function decorateLink(link, config, path) {
+  let pathname = link.getAttribute('href');
+  try { pathname = new URL(pathname).pathname; } catch (e) { /* href does not contain domain */ }
+  const linkParts = pathname.split('/');
+  const prefix = linkParts[1] || 'us';
+  let { href } = link;
+  if (href.endsWith('/')) href = href.slice(0, -1);
+  link.href = `${href}${config.contentRoot || ''}${path}`;
+  link.addEventListener('click', (e) => {
+    e.preventDefault();
+    handleEvent(prefix, link, config);
+  });
+}
+
+export default function decorate(block) {
+  const config = getConfig();
+  const regionSelectorBlock = block.querySelector('.region-selector');
+  const links = regionSelectorBlock?.querySelectorAll('a');
+  if (!links || !links.length) return;
```

Comment:
nit - can leave as is or use `if (!links?.length) return;`

Reply:
that's nicer - thx!